tiledb/ml/models/tensorflow.py

```diff
@@ -229,10 +229,13 @@ def _write_array(
 
             # Add extra metadata given by the user to array's metadata
             if meta:
-                for key, value in meta.items():
-                    tf_model_tiledb.meta[key] = json.dumps(
-                        value, default=json_utils.get_json_type
-                    ).encode("utf8")
+                self.update_model_metadata(array=tf_model_tiledb, meta=meta)
```

Comment:
Should we somehow protect the `self.file_properties`? What will happen if someone gives `meta` with same keys as the reserved keys for `file_properties`?

Reply:
Will do and will also add a check when the users provide extra metadata.